spring-batch-infrastructure/src/main/java/org/springframework/batch/support/PatternMatcher.java

```diff
@@ -202,12 +217,23 @@ public S match(String line) {
 		Assert.notNull(line, "A non-null key must be provided to match against.");
 
 		for (String key : sorted) {
-			if (PatternMatcher.match(key, line)) {
+			if (match(key, line)) {
 				value = map.get(key);
 				break;
 			}
 		}
 
+		if (value == null) {
+			for (String key : sorted) {
+				try {
+					if (matchRegex(key, line)) {
+						value = map.get(key);
+						break;
+					}
+				} catch (Throwable ignored) {}
+			}
+		}
```

Comment:
Maybe we can replace `match()` to `matchRegex()`, but it makes **breaking change**.
- e.g. `"*"` means matching all lines on `match()`, but invalid regex format and throws `PatternSyntaxException` on `matchRegex()`

(also PatternMatcherTests failed)

To avoid this breaking change, I intentionally `matchRegex()` only when there's nothing matched by `match()`~!